services/provider/server/consumer.go

```diff
@@ -83,7 +85,15 @@ func (c *ocsConsumerManager) Create(ctx context.Context, onboard ifaces.StorageC
 		},
 	}
 
-	err := c.client.Create(ctx, consumerObj)
+	onboardingTicket, err := util.DecodeOnboardingTicket(ticket)
+	if err != nil {
+		klog.Warningf("onboarding ticket has invalid quota %v :", err.Error())
+		return "", errTicketInvalidStorageQuota
+	}
+
+	consumerObj.Spec.StorageQuotaInGiB = int(onboardingTicket.StorageQuotaInGiB)
```

Comment:
Why are we decoding a ticket inside the storage consumer manager create method? We should accept the quota as a parameter for this method. 

I would expect the existing onboarding flow to already decode the ticket in order to validate it. We should use the decoded ticket from that step to get the quota.

Reply:
Yes, I was not sure, so I checked with Leela and chose this approach. Leela said, "Just try any one you like; we can discuss it over PR."